File: app/models/company.rb
# frozen_string_literal: true
# typed: strict

class Company < ApplicationRecord
  has_many :company_technologies, dependent: :destroy
  has_many :technologies, through: :company_technologies

  belongs_to :city
  has_one :region, through: :city
  has_one :country, through: :region
  has_one :continent, through: :country

  validates :name, presence: true
  validates :website, url: { no_local: true, public_suffix: true }
  validates :careers_page, url: { no_local: true, public_suffix: true }, allow_blank: true
end

File: app/frontend/types/company.ts
import { GeoFragment } from "./fragments/geo";
import { Technology } from "./technology";

export interface Company {
  id: number;
  slug: string;
  name: string;
  website?: string;
  careersPage?: string;
  description?: string;
  technologies: Technology[];
  address: {
    short: string;
    medium: string;
    full: string;
  };
  continent: GeoFragment;
  country: GeoFragment;
  region: GeoFragment;
  city: GeoFragment;
}
